src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -618,21 +618,45 @@ await foreach (ConfigurationSetting setting in client.GetConfigurationSettingsAs
                 }).ConfigureAwait(false);
             }
 
-                foreach (var loadOption in _options.KeyValueSelectors)
+            foreach (KeyValueSelector loadOption in _options.KeyValueSelectors)
+            {
+                IAsyncEnumerable<ConfigurationSetting> settingsEnumerable;
+
+                if (string.IsNullOrEmpty(loadOption.SnapshotName))
+                {
+                    settingsEnumerable = client.GetConfigurationSettingsAsync(
+                        new SettingSelector
+                        {
+                            KeyFilter = loadOption.KeyFilter,
+                            LabelFilter = loadOption.LabelFilter
+                        },
+                        cancellationToken);
+                }
+                else
                 {
-                    var selector = new SettingSelector
+                    ConfigurationSettingsSnapshot snapshot = await client.GetSnapshotAsync(loadOption.SnapshotName);
+
+                    if (snapshot != null)

```

Comment:
Perhaps we should require snapshot to exist as well.

@zhenlan

Reply:
So the SDK doesn't throw if a snapshot doesn't exist but throws if a key-value doesn't exist?

Reply:
Anyways, yes, it makes sense to throw if a snapshot doesn't exist.

Reply:
I expect consistent behavior between key-value and snapshot retrieval. @amerjusupovic can you validate?

Reply:
Yes, I'll test this to confirm the expected behavior

Reply:
I just overlooked it before, but SDK does throw. The snapshot null check shouldn't be necessary then.

Reply:
Right, no null check, but we do need to catch the exception and convert it to our exception.

Reply:
Ok, so do you mean instead of RequestFailedException change it something else, like ArgumentException? And I think I should be catching these exceptions in InitializeAsync then right? We wouldn't want these exceptions to stop the application if optional is true.

Reply:
> Ok, so do you mean instead of RequestFailedException change it something else, like ArgumentException? 

Yes that we shouldn't be throwing RequestFailedException, but I don't suggest ArgumentException. There is no invalid argument. The only argument the user is passing is a snapshot name which is valid. For lack of a better built-in option I'd go with InvalidOperationException, similarly to how we are doing for composition type.

Reply:
> And I think I should be catching these exceptions in InitializeAsync then right? We wouldn't want these exceptions to stop the application if optional is true.

Agreed